gammapy/datasets/io.py

```diff
@@ -472,3 +477,167 @@ def read(self):
+                "Energy true axes of exposure and DRM do not match. Check fermipy configuration."
+            )
+            edisp_axes = edisp.edisp_map.geom.axes
+
+        psf_r68s = psf.containment_radius(
+            0.68,
+            edisp_axes["energy_true"].center,
+            position=counts.geom.center_skydir,
+        )
+        # check that pdf is well defined (fails if edisp_bins>0 in fermipy)
+        if np.any(psf_r68s.value) == 0.0:
+            raise ValueError(
+                "PSF is not defined for all true energies. Check fermipy configuration."
+            )
+
+        # change counts energy axis unit keV->MeV
+        energy_axis = counts.geom.axes["energy"]._init_copy(
+            nodes=edisp_axes["energy"].edges
+        )
+        geom = counts.geom.to_image().to_cube([energy_axis])
+        counts = Map.from_geom(geom, data=counts.data)
+
+        # standardize dataset interpolating to same geom and axes
+        dataset = MapDataset(
+            counts=counts,
+            exposure=exposure,
+            psf=psf,
+            edisp=edisp,
+            name=name,
+        )
+        dataset = create_map_dataset_from_dl4(
+            dataset, geom=counts.geom, name=dataset.name
+        )
+
+        if edisp_bins > 0:  # slice edisp_bins
+            dataset = dataset.slice_by_idx(
+                dict(energy=slice(edisp_bins, -edisp_bins)), name=dataset.name
+            )
+
+        if isotropic_file:
+            model = create_fermi_isotropic_diffuse_model(
+                isotropic_file, datasets_names=[dataset.name]
+            )
+            dataset.models = Models([model])
+        return dataset
+
+    def read(self):
+        """Create Fermi-LAT map datasets from Fermipy configuration file.
+
+        Returns
+        -------
+        dataset : `~gammapy.datasets.Datasets`
+            Map datasets.
+
+        """
+        from gammapy.datasets import Datasets
+
+        filename = self.filename.resolve()
+        cwd = os.getcwd()
+        data = read_yaml(filename)
+        os.chdir(filename.parent)
+
+        if "components" in data:
+            components = data["components"]
+        else:
+            components = [data]
+
+        datasets = Datasets()
+        for file_id, component in enumerate(components):
+            if "fileio" in component and "outdir" in component["fileio"]:
+                path = component["fileio"]["outdir"]
+            elif "fileio" in data and "outdir" in data["fileio"]:
+                path = Path(data["fileio"]["outdir"])
+            else:
+                path = Path("")
+
+            if "model" in component and "isodiff" in component["model"]:
+                isotropic_file = Path(component["model"]["isodiff"])
+                name = isotropic_file.stem[4:]
+            elif "model" in data and "isodiff" in data["model"]:
+                isotropic_file = Path(data["model"]["isodiff"])
+                name = isotropic_file.stem[4:]
```

Comment:
This reading is tied strictly to the current fermipy file format?
We should have an info about the version we support.